src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -381,12 +381,18 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        WeakReference<String> internedString = stringCache.get(str);
-        if ( internedString == null || internedString.get() == null) {
-            internedString = new WeakReference<>(str);
-            stringCache.put(str, internedString);
+        if (!vcfIndexUseStringsCache)
+            return str;
+        WeakReference<String> stringWeakReference = stringCache.get(str);
```

Comment:
My main concern is that if we do this, I'd like there to be a test-target that covers this, but it would have to live in GATK or Picard and would then not be directly testing htsjdk, and it would be difficutl to test branches/PRs in htsjdk...also, it would be a long-running test and so should only be invoked "on occasion"...I'm not sure I know how to set something like this up....